src/main/scala/mesosphere/marathon/core/task/update/impl/steps/NotifyRateLimiterStepImpl.scala

```diff
@@ -28,9 +28,9 @@ class NotifyRateLimiterStepImpl @Inject() (
 
   override def process(update: InstanceChange): Future[Done] = {
     update.condition match {
-      case condition if limitWorthy(condition) =>
+      case condition if limitWorthy(condition) && update.stateUpdated =>
```

Comment:
out of curiosity: what does this change do?

Reply:
I am glad you asked! :)

RateLimiter is triggered for every `InstanceChange` event. Right now, `InstanceChange` is very tight to condition change so InstanceChange pretty much was condition change all the time. But now, we will be having `InstanceChange` event also for goal changes, so my idea was not to trigger rate limiter for that

Reply:
so you only trigger rate limiter if the underlying condition actually change. I see. That definitely deserve an explanatory comment:
```
// RateLimiter is triggered for every InstanceChange event. Right now, InstanceChange is very tight to condition 
// change so InstanceChange pretty much was condition change all the time. But now, we will be having 
// InstanceChange event also for goal changes, so my idea was not to trigger rate limiter for that
```